src/stories/tag/Tag.tsx

```diff
@@ -1,13 +1,21 @@
-import type { ComponentPropsWithoutRef, FC, ReactNode } from 'react';
+import type {
+    ComponentPropsWithoutRef, FC, ReactElement, ReactNode,
+} from 'react';
 import clsx from 'clsx';
 import styles from './Tag.module.scss';
 import typography from '../text/Typography.module.css';
+import { RemoveFromDOM } from '../utility';
+import { Icon, Check } from '../icon';
 
 export type TagProps = {
     /** The size of the Tag. */
-    size?: 'normal' | 'compact';
+    size?: 'normal' | 'compact' | 'small';
```

Comment:
please rename as discussed

i would also def encourage separating this into a separate prop of some kind honestly — especially because there are probably instances where we might want to customize size separately for an icon tag

since compact and "small"/icon have the same height, it would make more sense to expect `size="compact" shape="square"` or something like that for a 20px icon tag (similar to how we handle buttons)

eventually we could potentially change `size` components like this to also accept a pixel quantity for custom sizing, so I think separating out icon (which is more of a shape/format/etc anyways than a size) makes sense

Reply:
if there is no situation where a tag has an icon AND label visible, then you could potentially also just refer to the `icon` prop for whether a tag is an icon tag vs a normal tag. `icon={null}` means normal tag, passing an icon prop means it's an icon tag.

I think I'd still prefer the former dx of a separate prop for defining when an icon

Reply:
If you really want to emulate the way we handle buttons, then should we also separate the styles from "normal" and "compact"? Right now "normal" is taller, plus with a rectangle shape and border. "Compact" is shorter, plus with a rounded rectangle and no border.

We could do `size="normal"` vs `size="compact"` to change only the height. Then `shape="rectangle"` would have the current "normal" style of rectangle with border, `shape="rounded"` would have the current "compact" style of rounded rectangle no border, and `shape="square"` would be a rounded square with icon inside.